cli/root.go

```diff
@@ -93,7 +95,19 @@ func NewAlertmanagerClient(amURL *url.URL) *client.Alertmanager {
 		cr.DefaultAuthentication = clientruntime.BasicAuth(amURL.User.Username(), password)
 	}
 
-	return client.New(cr, strfmt.Default)
+	c := client.New(cr, strfmt.Default)
+
+	status, err := c.General.GetStatus(nil)
+	if err != nil || status.Payload.VersionInfo == nil || version.Version == "" {
+		// We can not get version info, or we do not know our own version. Let amtool continue.
```

Comment:
Now that we have an option to cancel the check, I don't think we should fail silently. How do you feel about logging that we were unable to compute the version and tell the user that should upgrade/downgrade?

Reply:
If we can not run the check, it is because alertmanager does not work at all, and the error will be printed in the actual call later on.

Reply:
I was thinking more in the [case of Grafana](https://github.com/grafana/grafana/blob/b78a67cec745b11b05ba2cf7ff12d848798b07a5/pkg/services/ngalert/api/tooling/definitions/alertmanager.go#L178-L197) but this is not a concern for us here to be honest.